R/themes.R

```diff
@@ -1,10 +1,54 @@
-#' @title Themes
+#' @title Theme
 #' @docType class
 #' @description
 #' Theme class sets of font properties for texts
 #' @include utils.R
 #' @export
-Themes <- R6::R6Class(
-  "Themes",
-  public = list()
+Theme <- R6::R6Class(
+  "Theme",
+  public = list(
+    title = NULL,
+    subtitle = NULL,
+    xlabel = NULL,
+    ylabel = NULL,
+    watermark = NULL,
+
+    initialize = function(colorPalette = labelColorPalette$default, baseSize=12) {
+      self$title <- Font$new(color = colorPalette$title, size = baseSize * 2, fontFace = "bold")
+      self$subtitle <- Font$new(color = colorPalette$subtitle, size = baseSize * 1.5, fontFace = "italic")
+      self$xlabel <- Font$new(color = colorPalette$xlabel, size = baseSize)
+      self$ylabel <- Font$new(color = colorPalette$ylabel, size = baseSize)
+      self$watermark <- Font$new(color = colorPalette$watermark, size = baseSize)
+    },
+
+    setPlotConfiguration = function(PlotConfiguration) {

```

Comment:
I'd move this guy to the PlotConfiguration object. (instead of havign the them update the plotconfiguration it should be the plitConfiguration updating itselfs from the theme passed as parameter.
Same code, just upside down